app/sample_manifest_excel/sample_manifest_excel/upload/row.rb

```diff
@@ -87,8 +87,9 @@ def update_sample(tag_group, override) # rubocop:todo Metrics/AbcSize
         if sample.updated_by_manifest && !override
           @sample_skipped = true
         else
-          update_specialised_fields(tag_group)
+          update_late_specialised_fields(tag_group)
           asset.save!
+          update_metadata_fields
```

Comment:
Trying to unpick this one, as it also appears below, as part of sample validation. (Which makes sense, as you can't validate the data before its set) Why is it also called here?

Reply:
This was the commit - https://github.com/sanger/sequencescape/pull/3231/commits/e7ba7cb30bab000720bc93e7d27a8f362c1ec00e - think the message explains it OK.

I think it's only actually necessary for the test, as it gets set as part of the sample validation as you say. Do you think I should rewrite the test so it's not needed, and take it out here?

Reply:
Probably, but let's leave it for now. As you said, the commit message explains it pretty well. (Sorry for missing it)